src/toolbars/DeviceToolBar.cpp

```diff
@@ -351,42 +452,44 @@ void DeviceToolBar::EnableDisableButtons()
       // we allow changes when monitoring, but not when recording
       bool audioStreamActive = gAudioIO->IsStreamActive() && !gAudioIO->IsMonitoring();
 
-      // Here we should relinquish focus
       if (audioStreamActive) {
-         wxWindow *focus = wxWindow::FindFocus();
-         if (focus == mHost || focus == mInput || focus == mOutput || focus == mInputChannels)
-            TrackPanel::Get( mProject ).SetFocus();
+         mOpenAudioSetup->Disable();
+      }
+      else {
+         mOpenAudioSetup->Enable();
       }
-
-      mHost->Enable(!audioStreamActive);
-      mInput->Enable(!audioStreamActive);
-      mOutput->Enable(!audioStreamActive);
-      mInputChannels->Enable(!audioStreamActive);
    }
 }
 
-void DeviceToolBar::SetNames()
-{
-   /* i18n-hint: (noun) It's the device used for playback.*/
-   mOutput->SetName(_("Playback Device"));
-   /* i18n-hint: (noun) It's the device used for recording.*/
-   mInput->SetName(_("Recording Device"));
-   mHost->SetName(_("Audio Host"));
-   mInputChannels->SetName(_("Recording Channels"));
-}
-
 void DeviceToolBar::RegenerateTooltips()
 {
 #if wxUSE_TOOLTIPS
-   SetNames();
-   mOutput->SetToolTip(mOutput->GetName() + wxT(" - ") + mOutput->GetStringSelection());
-   mInput->SetToolTip(mInput->GetName() + wxT(" - ") + mInput->GetStringSelection());
-   mHost->SetToolTip(mHost->GetName() + wxT(" - ") + mHost->GetStringSelection());
-   mInputChannels->SetToolTip(mInputChannels->GetName() + wxT(" - ") + mInputChannels->GetStringSelection());
+   for (long iWinID = ID_OPEN_AUDIO_SETUP_BUTTON; iWinID < BUTTON_COUNT; iWinID++)
+   {
+      auto pCtrl = static_cast<AButton*>(this->FindWindow(iWinID));
+      CommandID name;
+      switch (iWinID)
+      {
+      case ID_OPEN_AUDIO_SETUP_BUTTON:
+         name = wxT("Open Audio Setup");
+         break;
+      }
+      std::vector<ComponentInterfaceSymbol> commands(
+         1u, { name, Verbatim(pCtrl->GetLabel()) });
+
+      // Some have a second
+      switch (iWinID)
+      {
+      case ID_OPEN_AUDIO_SETUP_BUTTON:
+         break;
+      }
+      ToolBar::SetButtonToolTip(
+         mProject, *pCtrl, commands.data(), commands.size());
+   }
```

Comment:
I see more imitation of ControlToolBar.cpp here.  Do you understand what this is doing?

For ControlToolBar, it looks up `name` in the CommandManager to determine the keystroke shortcut, if any, that is associated with that name, and then formats the hover text for the button to include that shortcut key in the description.

Here, no key will be found because "Open Audio Setup" is not the name of any command.

But then, perhaps it should be registered as a named command so it can indeed be bound to a keystroke if the user chooses in Preferences -- even if we do not decide to give it a key binding by default.

@DavidBailes would be interested I'm sure in keystroke accessibility of the new menu under the cogwheel.